### PR TITLE
Add ShockScript highlighting

### PR DESCRIPTION
**Pull request by Hydroper**
## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [x] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Comment by Alhadis** [comment]
@Hydroper Please read [our guidelines on adding a new language](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language), and fill out the pull-request template. Make sure to read the requirements a language must fulfil to qualify for recognition by Linguist.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -6285,6 +6285,15 @@ sed:
...
   language_id: 847830017
+ShockScript:
+  type: programming
+  color: "#ff9070"
+  extensions:
+  - '.sx'
+  - '.shockscript'
+  tm_scope: source.sx
+  ace_mode: actionscript
+  language_id: 850000000
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Please follow the guidelines to add a new language. This id shouldn't be set manually.

File: grammars.yml [hunk 2]
```diff
@@ -872,3 +872,5 @@ vendor/grammars/zeek-sublime:
 - source.zeek
 vendor/grammars/zephir-sublime:
 - source.php.zephir
+vendor/grammars/shockscript-highlighting:
+- source.sx
```
Comment (pchaigno) [inline grammars.yml]: Please use the provided script as per the guidelines to add a new grammar.

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -6285,6 +6285,15 @@ sed:
   ace_mode: text
   tm_scope: source.sed
   language_id: 847830017
+ShockScript:
+  type: programming
+  color: "#ff9070"
+  extensions:
+  - '.sx'
+  - '.shockscript'
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: The pull request is missing sample files for these extensions.

**Pull request closed without merging**